### DISCUSSION TITLE
Extrude until

### DISCUSSION THREAD

**Discussion by hrgdavor** [elalish/manifold#941] [Ideas]
I noticed an interesting operation in build123 (I guess it is from OCCT). I am wondering if such operation is possible with manifold already, and if not if it would be beneficial (for performance or at least for utility).

Here is description from docs:

[36. Extrude Until](https://build123d.readthedocs.io/en/latest/introductory_examples.html#extrude-until)
Sometimes you will want to extrude until a given face that could be non planar or where you might not know easily the distance you have to extrude to. In such cases you can use [extrude()](https://build123d.readthedocs.io/en/latest/operations.html#operations_part.extrude) [Until](https://build123d.readthedocs.io/en/latest/builder_api_reference.html#build_enums.Until) with Until.NEXT or Until.LAST.

![image](https://github.com/user-attachments/assets/a61fe77e-ee25-4ef8-bc09-53b346a0c5a4)

**Comment by elalish** [comment]
I've occasionally thought about an algorithm like this, because it would be quite convenient sometimes. However, I can't think of a clear mathematical way to describe what the answer should be in general (way before figuring out an algorithm to accomplish that). What happens when the cross section doesn't run into anything? What about when it runs into something for half of the cross section? 

Maybe it would work if we made an `ExtrudeUnion` function and keep our extrude parameters (with a height)? I think for it to make any sense, that would mean if the cross section does not *entirely* run into an object, it would be the same as `Extrude(a).Union(b)`, but if it did entirely intersect, then it could stop early. An in-between issue is when the object it intersects curves back on itself so the cross section hits it in two different places. Geometrically it might be possible, but probably too tricky to write a robust algorithm for.

It would be pretty non-trivial to implement. I think it would be cool, but I'd put it a lot further down the priority list than most of our other ideas unless a whole bunch of folks need it.

**Reply by hrgdavor** [reply]
More I think of it, it really would have to bring significant improvement over `Extrude(a).Union(b)` to warrant special case implementation. 

ATM it looks just like a neat idea, I can not give some strong reason to have it.
